indigo_api/views/documents.py

```diff
@@ -463,3 +463,39 @@ def get_serializer(self, queryset, *args, **kwargs):
             serializer.data[i]['_snippet'] = doc.snippet
 
         return serializer
+
+
+class ComparisonView(APIView):
+    """ Support for running a document comparison.
+    """
+    permission_classes = (IsAuthenticated,)
+
+    def post(self, request):
+        serializer = DocumentAPISerializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+
+        differ = AttributeDiffer()
+
+        current_document = serializer.fields['document'].update_document(Document(), serializer.validated_data['document'])
+        current_document.document_xml = differ.preprocess_document_diff(current_document.document_xml)
+        current_html = current_document.to_html()
+
+        comparison_doc_id = request.data['comparison_doc_id']
+        comparison_document = Document.objects.get(id=comparison_doc_id)
```

Comment:
if comparison_doc_id is bad and the document doesn't exist, raise an `Http404()`

Reply:
`objects.get` raises `Document.DoesNotExist` if the document doesn't exist.